dev/com.ibm.ws.security.javaeesec.cdi/src/com/ibm/ws/security/javaeesec/cdi/extensions/JavaEESecCDIExtension.java

```diff
@@ -316,13 +340,47 @@ private void createModulePropertiesProviderBeanForBasicToAdd(BeanManager beanMan
         }
     }
 
+    /**
+     * @param annotations
+     * @param props
+     */
+    private void addDecoratorAlternativeProps(Set<Annotation> annotations, Properties props) {
+        //This is class level annotation
+        for (Annotation annt : annotations) {
+            Class<? extends Annotation> annType = annt.annotationType();
+            //Tr.debug(tc, "Decorated present: " + annType.isAnnotationPresent(Decorated.class));
+            if (annType.getName().equals("jakarta.decorator.Decorator")) {
+                if (props == null) {
+                    props = new Properties();
+                }
+                if (tc.isDebugEnabled())
+                    Tr.debug(tc, "Add Decorator property");
+                props.put(decorator, true);
+                break;
+            }
+        }
+    }
+
+    private boolean isDecorator(Set<Annotation> annotations) {
+        for (Annotation annotation : annotations) {
+            Class<? extends Annotation> annotationType = annotation.annotationType();
+            if (annotationType.getName().equals("jakarta.decorator.Decorator")) {
+                return true;
+            } else if (annotationType.getName().equals("jakarta.enterprise.inject.Alternative")) { //TODO: UTLE - right annotation?
```

Comment:
That is the correct annotation 👍  but if we want to support the behavior for appSecurity-3.0(javaee8) then we need to check for the javax package name too `javax.enterprise.inject.Alternative`.
```suggestion
            } else if (annotationType.getName().equals("jakarta.enterprise.inject.Alternative") || annotationType.getName().equals("javax.enterprise.inject.Alternative")) { 
```